app/models/ext_management_system.rb

```diff
@@ -482,15 +484,29 @@ def vm_count_by_state(state)
     vms.inject(0) { |t, vm| vm.power_state == state ? t + 1 : t }
   end
 
-  def total_vms_on;        vm_count_by_state("on");        end
+  def host_count_by_metric(metric)
+    total_metric = hosts.inject(0) { |t, host| t + host.send(metric) }
```

Comment:
Thanks for pinging me @gtanzillo - this was prime for a `virtual_column` (and even better, we could use a `virtual_aggregate` which wrote all the ruby code for us as well)